Adding info on updating integrations (#20052)
DataDog/documentation: content/en/developers/integrations/agent_integration.md

Once you have created your Agent-based integration, see the [Create a tile][20] documentation for information on populating the remaining [required assets][31] that appear on your integration tile, and opening a pull request.

## Update your integration
To update your integration, edit the relevant files and open a new pull request to your integration's directory in the [`integrations-extras`][21] or [`marketplace`][22] repository. 

> * If you are editing or adding new integration code, a version bump is required.

Reviewer: Do you want to mention the diff between major and minor version bumps?

Author: discussed offline - going to resolve

> * If you are editing or adding new README content, manifest information, or assets such as dashboards and recommended monitors, a version bump is optional. 

Reviewer 2: Nit: version bumps for assets and README are unnecessary, they would be bumping the version but nothing would change. I think we should say these changes don't need a version bump instead of being optional.

Reviewer 3: +1

Author: Gotcha, will change this

> With or without a version bump, updates to assets such as dashboards and recommended monitors immediately become available to customers after the corresponding pull requests are merged and they are published. Updates to `README.md`, `manifest.json`, or any other non-code files also immediately become visible to customers after publishing. 

Reviewer 4: Its not necessarily `immediate`, but there's indeed no customer or developer intervention needed to see these updates.  

```suggestion
With or without a version bump, updates to assets such as dashboards and recommended monitors automatically become available to customers after the corresponding pull requests are merged and they are published. Updates to `README.md`, `manifest.json`, or any other non-code files also immediately become visible to customers after publishing. 
```

> With or without a version bump, updates to assets such as dashboards and recommended monitors are available to customers after the corresponding pull requests are merged and the assets are published. Updates to `README.md`, `manifest.json`, or any other non-code files are also immediately available to customers after publishing. 

Reviewer 3: Is this true? I know that our team syncs assets once per week. Naturally if something is urgent, that gets synced separately, but I'm not sure that's the same expectation as what this sentence sets up.

Author: @nmuesch knows better than me here

Reviewer 3: my two cents would be to make point 2 more explicit. You can literally say something along the lines of your comment: "no more action needed from you [the contributor], we take care of it".

As for point 1: our team syncs assets individually for cases where we know there are changes. For instance releases of new integrations or urgent fixes to assets that we make ourselves and that somebody's waiting for. The weekly sync captures non-urgent changes and/or changes by external contributors to core and extras repos.

Author: @iliakur @nmuesch made some updates based on this comment thread. Let me know what you think

Reviewer 3: lgtm

Reviewer 4: Agreed, the updates look good to me too!

Author: Great! @alai97 can we merge plz?

> With or without a version bump, updates to assets such as dashboards and recommended monitors automatically become available to customers after the corresponding pull requests are merged and they are published. Updates to `README.md`, `manifest.json`, or any other non-code files also immediately become visible to customers after publishing. 

Reviewer 5: ```suggestion
With or without a version bump, updates to assets such as dashboards and recommended monitors are available to customers after the corresponding pull requests are merged and the assets are published. Updates to `README.md`, `manifest.json`, or any other non-code files are also immediately available to customers after publishing. 
```

### Bumping an integration version 
In addition to any code changes, the following is required when bumping an integration version:

> 1. Update `__about__.py` to reflect the new version number. This file can be found in your integration directory under `/datadog_checks/<your_check_name>/__about__.py`.

Reviewer 5: ```suggestion
1. Update `__about__.py` to reflect the new version number. This file can be found in your integration's directory under `/datadog_checks/<your_check_name>/__about__.py`.
```

2. Add an entry to the CHANGELOG.md file that adheres to the following format:
   ```
   ## Version Number / Date

   ***Added***: 

   * New feature
   * New feature

   ***Fixed***:

   * Bug fix
   * Bug fix
   ```

> 3. Update all references to the version number where mentioned in `README.md` and elsewhere. Oftentimes installation instructions in `README.md` include the version number, which needs to be updated.

Reviewer 5: ```suggestion
3. Update all references to the version number mentioned in `README.md` and elsewhere. Installation instructions in `README.md` often include the version number, which needs to be updated.
```
